From 7af6650c2c267be3f95e57933db4d2de91581ecc Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Wed, 2 Dec 2020 13:59:53 +0700
Subject: [PATCH 288/358] arm64: dts: renesas: r8a774a3-hihope-rzg2m: Add
 reserved memory regions

This commit adds reserved memory region for:
- Lossy Decompress: fixed address defined in IPL and mmngr.
- Global CMA: 512MiB from 0x0058000000 to 0x0077ffffff
- MMP CMA: 256MiB from 0x0078000000 to 0x0087ffffff

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 .../boot/dts/renesas/r8a774a3-hihope-rzg2m.dts     | 27 ++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts b/arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts
index 987c1f0..f138294 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts
@@ -23,4 +23,31 @@
 		device_type = "memory";
 		reg = <0x6 0x00000000 0x0 0x80000000>;
 	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		/* device specific region for Lossy Decompression */
+		lossy_decompress: linux,lossy_decompress@54000000 {
+			no-map;
+			reg = <0x00000000 0x54000000 0x0 0x03000000>;
+		};
+
+		/* global autoconfigured region for contiguous allocations */
+		global_cma: linux,cma@58000000 {
+			compatible = "shared-dma-pool";
+			reusable;
+			reg = <0x00000000 0x58000000 0x0 0x20000000>;
+			linux,cma-default;
+		};
+
+		/* device specific region for contiguous allocations */
+		mmp_reserved: linux,multimedia@70000000 {
+			compatible = "shared-dma-pool";
+			reusable;
+			reg = <0x00000000 0x78000000 0x0 0x10000000>;
+		};
+	};
 };
-- 
2.7.4

